src/main/java/com/example/manager/controller/UserController.java

```diff
@@ -0,0 +1,143 @@
+package com.example.manager.controller;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.example.manager.repository.UserRepository;
+
+import com.example.manager.domain.User;
+
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Service
+public class UserController {
+
+    public static final int maxFailedAttempts = 3;
+
+    // 60 Seconds
+    private static final long lockTimeDuration = 60 * 1000;
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Autowired
+    private UserRepository repo;
+
+    final private PasswordEncoder passwordEncoder = passwordEncoder();
+
+    public UserController() {
+        super();
+    }
+
+
+    public User register(User user) {
+        return repo.save(user);
+    }
+
+    public boolean checkIfUsernameTaken(String username) {
+        return repo.existsById(username);
+    }
+
+
+    public boolean login(String username, String password) {
+        User temp = null;
+        try {
+            temp = repo.findById(username).get();
+        } catch (Exception e) {
+
+        }
+        if (temp != null) {
+            if (passwordEncoder.matches(password, temp.getPassword()) && temp.getAccountNonLocked()) {
+                resetFailedAttempts(username);
+                return true;
+            } else {
+                unlockAfterTimeExpired(username);
+                temp = repo.findById(username).get();
+                if (temp.getAccountNonLocked() && temp.getFailedAttempt() < maxFailedAttempts) {
+                    increaseFailedAttempts(username);
+                    temp = repo.findById(username).get();
+                    System.out.println("Failed attempt " + temp.getFailedAttempt() + "/4.");
+                } else {
+                    temp = repo.findById(username).get();
+                    if (temp.getFailedAttempt() == maxFailedAttempts) {
+                        lock(username);
+                        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+                        LocalDateTime  loginFailed = LocalDateTime.now().plusSeconds(60);
+                        System.out.println("User locked until " + dtf.format(loginFailed) + ".\n");
+                        return false;
+                    }

```

Comment:
In line 52 you try to fetch the Db User. If its not null you have the actual entity. However, later you ask the DB again for the user. This is not need.

Also your "login" method has some side effects. 
I suggest you extract the code in smaller methods.

e.g.: 
handleFailedPasswordAttempt(user);
and
lockUser(user);